### PR TITLE
Allow CSS Code Hints in PHP

### LINKED ISSUE #11629: Code hint/completion regardless of extension
I don't know if this is an error or as intended but I can not under any circumstances get code completion for my css to appear if the file extension is .php.

It will complete the tag... "&lt;style&gt;" "&lt;/style&gt;" but if I add a class and begin entering style definitions, the popup window with suggestions never appears... however If I simply change the file extension to "html" and start typing the same style under my class it works as expected.

I believe this may be an error as it works correctly inside javascript tags regardless if the extension is PHP. Whether or not this is an error it would be nice to see it resolved so CSS would auto-complete regardless of file extension, Furthermore I have tried many different preference changes to no avail but if there is some workaround now that would be great.

### PR DESCRIPTION
**Pull request by sprintr**
Allow CSS Code Hints in style tags for PHP files. Fixes #11629

CC: @abose

### CODE REVIEW

File: src/language/CSSUtils.js [hunk 1]
```diff
@@ -83,19 +83,31 @@ define(function (require, exports, module) {
     
     /**
      * @private
+     * Returns state of a token
+     * @param {{editor:{CodeMirror}, pos:{ch:{string}, line:{number}}, token:{object}}} ctx
+     * @return {{tokenize:function, state:string, stateArg:string, context:Object}}
+     */
+    function _getState(ctx) {
```
Comment (ficristo) [inline src/language/CSSUtils.js]: Can you rename it like _getTokenState or _getLocalState? _getState seems too common to me.
Can you add an assert for ctx?

File: src/language/CSSUtils.js [hunk 2]
```diff
@@ -83,19 +83,31 @@ define(function (require, exports, module) {
...
      * @param {editor:{CodeMirror}, pos:{ch:{string}, line:{number}}, token:{object}} context
      * @return {boolean} true if the context is in property name
      */
     function _isInPropName(ctx) {
-        var state,
+        var state = _getState(ctx),
             lastToken;
-        if (!ctx || !ctx.token || !ctx.token.state || ctx.token.type === "comment") {
+        if (!state || !ctx || !ctx.token || ctx.token.type === "comment") {
             return false;
```
Comment (ficristo) [inline src/language/CSSUtils.js]: Here, and twice below, you should move the check at the end of the `if` (after you are sure ctx is valid)
or simply move the assignment after the first `if`.

**Comment by abose** [comment]
Hi guys, As we will be releasing 1.5 by Thursday, this change has to be pulled in by Tuesday or we could push this to 1.6.

**Comment by sprintr** [comment]
@abose Cool! Will push the changes in a while.

File: src/language/CSSUtils.js (hunk 1 above)
Comment (sprintr) [inline src/language/CSSUtils.js]: `getContextState` makes more sense here.

**Comment by abose** [comment]
Thanks @sprintr .
Verified, all unit tests passing. Merging :+1:

**Pull request merged**